### PR TITLE
monero: match monero's stricter check when decompressing points

### PR DESCRIPTION
**Pull request by j-berman**
https://github.com/serai-dex/serai/issues/102#issuecomment-1627474637
https://github.com/serai-dex/serai/issues/102#issuecomment-1627475063

### CODE REVIEW

File: coins/monero/src/wallet/send/mod.rs [hunk 1]
```diff
@@ -787,7 +787,7 @@ impl SignableTransaction {
       fee -= output.commitment.amount;
       tx_outputs.push(Output {
         amount: None,
-        key: output.dest.compress(),
+        key: output.dest.compress().to_bytes(),
```
Comment (kayabaNerve) [inline coins/monero/src/wallet/send/mod.rs]: Why did you remove the typing from CompressedEdwardsY to [u8; 32]?

Comment (j-berman) [reply coins/monero/src/wallet/send/mod.rs]: Reverted

**Comment by kayabaNerve** [comment]
LGTM on being consistent :)
